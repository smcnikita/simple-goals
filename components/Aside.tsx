'use client';

import { useMemo, type FC } from 'react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';

import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
} from '@/components/ui/sidebar';

import type { Years } from '@/types/years.types';

type Props = {
  years: Years;
};

const Aside: FC<Props> = ({ years }) => {
  const pathname = usePathname();

  const yearsLinks = useMemo(() => {
    return years.map((year) => ({
      title: year,
      url: `/goals/${year}`,
    }));
  }, [years]);

  const selectedYear = useMemo(() => {
    const segments = pathname.split('/');
    return Number(segments[2]);
  }, [pathname]);

  return (
    <Sidebar className="left-auto">
      <SidebarContent>
        <SidebarGroup>
          <SidebarGroupLabel>Years</SidebarGroupLabel>
          <SidebarGroupContent>
            <SidebarMenu>
              {yearsLinks.map((item) => (
                <SidebarMenuItem key={item.title}>
                  <SidebarMenuButton asChild isActive={selectedYear === Number(item.title)}>
                    <Link href={item.url}>
                      <span>{item.title}</span>
                    </Link>
                  </SidebarMenuButton>
                </SidebarMenuItem>
              ))}
            </SidebarMenu>
          </SidebarGroupContent>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
};

export default Aside;
