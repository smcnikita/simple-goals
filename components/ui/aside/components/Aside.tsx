import type { FC } from 'react';

import type { YearItem } from '@/types/years';

import AsideList from './AsideList';

import classes from '../styles/aside.module.css';

type Props = {
  years: YearItem[];
};

const Aside: FC<Props> = ({ years }) => {
  return (
    <aside className={classes.aside}>
      <nav>
        <AsideList years={years} />
      </nav>
    </aside>
  );
};

export default Aside;
