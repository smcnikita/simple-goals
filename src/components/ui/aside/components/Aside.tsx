import { type FC } from 'react';

import AsideList from './AsideList';

import classes from '../styles/aside.module.css';

const Aside: FC = () => {
  const currentYear = new Date().getFullYear();
  const years = [currentYear];

  return (
    <aside className={classes.aside} id="aside">
      <nav>
        <AsideList defaultYears={years} />
      </nav>
    </aside>
  );
};

export default Aside;
