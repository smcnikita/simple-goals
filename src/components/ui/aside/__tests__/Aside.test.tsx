import { render } from '@testing-library/react';

import Aside from '../components/Aside';

jest.mock('../components/AsideList', () => {
  const MockAsideList = (props: { defaultYears: number[] }) => (
    <div data-testid="aside-list-mock">{`Mocked AsideList with ${props.defaultYears.length} items`}</div>
  );
  MockAsideList.displayName = 'MockAsideList';
  return MockAsideList;
});

describe('Aside', () => {
  // TODO: add more tests

  it('should renders Aside unchanged', () => {
    const { container } = render(<Aside />);
    expect(container).toMatchSnapshot();
  });
});
