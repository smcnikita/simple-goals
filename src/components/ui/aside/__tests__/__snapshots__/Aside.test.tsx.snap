// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside should renders Aside unchanged 1`] = `
<div>
  <aside
    class="aside"
    id="aside"
  >
    <nav>
      <div
        data-testid="aside-list-mock"
      >
        Mocked AsideList with 1 items
      </div>
    </nav>
  </aside>
</div>
`;
